engine/src/flutter/impeller/entity/contents/filters/runtime_effect_filter_contents.cc

```diff
@@ -46,11 +47,54 @@ std::optional<Entity> RuntimeEffectFilterContents::RenderFilter(
   if (!input_snapshot.has_value()) {
     return std::nullopt;
   }
+
   std::optional<Rect> maybe_input_coverage = input_snapshot->GetCoverage();
   if (!maybe_input_coverage.has_value()) {
     return std::nullopt;
   }
-  Rect input_coverage = maybe_input_coverage.value();
+
+  // If the input snapshot does not have an identity transform the
+  // ImageFilter.shader will not correctly render as it does not know what the
+  // transform is in order to incorporate this into sampling. We need to
+  // re-rasterize the input snapshot so that the transform is absorbed into the
+  // texture.
+  // We can technically render this only when the snapshot is just a translated
+  // version of the original. Unfortunately there isn't a way to test for that
+  // though. Blur with low sigmas will return a transform that doesn't scale but
+  // has a tiny offset to account for the blur radius. That's indistinguishable
+  // from `ImageFilter.compose` which slightly increase the size to account for
+  // rounding errors and add an offset. Said another way; ideally we would skip
+  // this branch for the unit test `ComposePaintRuntimeOuter`, but do it for
+  // `ComposeBackdropRuntimeOuterBlurInner`.
+  if (!input_snapshot->transform.IsIdentity()) {
+    Matrix inverse = input_snapshot->transform.Invert();
+    Quad quad = inverse.Transform(Quad{
+        coverage.GetLeftTop(),     //
+        coverage.GetRightTop(),    //
+        coverage.GetLeftBottom(),  //
+        coverage.GetRightBottom()  //
+    });
+    TextureContents texture_contents;
+    texture_contents.SetTexture(input_snapshot->texture);
+    std::optional<Rect> bounds =
+        Rect::MakePointBounds(quad.begin(), quad.end());
+    if (bounds.has_value()) {
+      texture_contents.SetSourceRect(bounds.value());
+      texture_contents.SetDestinationRect(coverage);
+      texture_contents.SetStencilEnabled(false);
+      texture_contents.SetSamplerDescriptor(input_snapshot->sampler_descriptor);
```

Comment:
Shouldn't the sampler be nearest-neighbor? The input snapshots sampler will then be applied when the texture is composited up?